### PR TITLE
Use separate cache config for I18n.

### PR DESCRIPTION
**Pull request by ADmad**
Refs #5472

### CODE REVIEW

File: src/I18n/I18n.php [hunk 1]
```diff
@@ -70,7 +71,11 @@ public static function translators()
         );
 
         if (class_exists('Cake\Cache\Cache')) {
-            static::$_collection->setCacher(Cache::engine('_cake_core_'));
+            try {
+                static::$_collection->setCacher(Cache::engine('_cake_i18n_'));
+            } catch (InvalidArgumentException $e) {
+                static::$_collection->setCacher(Cache::engine('_cake_core_'));
+            }
```
Comment (markstory) [inline src/I18n/I18n.php]: Clever way to get backwards compatibility and let people who need the separate cache use it.

File: tests/bootstrap.php [hunk 2]
```diff
@@ -85,6 +85,11 @@
         'engine' => 'File',
         'prefix' => 'cake_model_',
         'serialize' => true
+    ],
+    '_cake_i18n_' => [
```
Comment (markstory) [inline tests/bootstrap.php]: Should we be adding this to the app template as well?

Comment (ADmad) [inline tests/bootstrap.php]: PR for app also done.

**Comment by lorenzo** [comment]
Do we use the _cake_core_ key for anything else?

**Comment by ADmad** [comment]
@lorenzo Good question :)  It seems in 3.0 `_cake_core_` is only used for i18n message caching. There no caching of paths and objects in 3.0 like we had in 2.x.

**Comment by lorenzo** [comment]
Does it make sense to "separate" this?

**Comment by ADmad** [comment]
@lorenzo Probably not, I didn't realize `_cake_core_` wasn't used for anything else anymore in 3.0. Though issue would come up if in future we do use `_cake_core_` to cache other things.

@markstory already merged the related PR I made for app skeleton :)  So that would have to be reverted in case we decide not to merge this.

**Comment by markstory** [comment]
Whoops. It is only used for translations in 3.x. However, in 2.x `_cake_core_` gets used for all sorts of things. I'll revert the app skeleton change.

**Pull request closed without merging**